lib/mdl/rules.rb

```diff
@@ -796,3 +796,85 @@ def check_blockquote(errors, elements)
     error_lines
   end
 end
+
+rule 'MD055', 'Tables should be well-formed' do
+  tags :tables
+  aliases 'table-format-validation'
+  check do |doc|
+    error_lines = []
+    tables = doc.find_type_elements(:table)
+    lines = doc.lines
+    num_lines = lines.length
+
+    tables.each do |table|
+      loc = table.options[:location]-1
+      header = lines[loc]
+
+      if header[0] != '|' || header[-1] != '|' || header.length == 1
+        # Tables should start and end with '|'
+        error_lines << (loc+1)
+      end
+
+      headings = header.strip().split('|').select{ |x| x.length > 0 }
```

Comment:
In ruby, it's idiomatic to drop the parens if there's no arguments and to put a space between a method and its block.
```suggestion
      headings = header.strip.split('|').select { |x| x.length > 0 }
```

Reply:
also, you can do `!x.empty?` if you prefer, a bit cleaner.